src/groups/bmq/bmqp/bmqp_messageproperties.cpp

```diff
@@ -300,19 +316,38 @@ bool MessageProperties::streamInPropertyValue(const Property& p) const
     }
 
     case bmqt::PropertyType::e_STRING: {
-        bsl::string value(p.d_length, ' ');
-        rc = bmqu::BlobUtil::readNBytes(&value[0],
-                                        d_blob.object(),
-                                        position,
-                                        p.d_length);
+        // Try to avoid copying the string.  'd_blop' already keeps a copy.
+        bmqu::BlobPosition end;
+        const int          ret =
+            bmqu::BlobUtil::findOffset(&end, *d_blob_p, position, p.d_length);
+        bool doCopy = true;
+        if (ret == 0) {
+            // Do not align
+            if (position.buffer() == end.buffer() ||
+                (position.buffer() + 1 == end.buffer() && end.byte() == 0)) {
```

Comment:
I think it's worth extracting this to `bmqu::BlobUtil::isDataContinuous(start, end)`